onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -1367,17 +1367,14 @@ def aten_constant_pad_nd(self: TTensor, pad: INT64, value: float = 0.0) -> TTens
     return op.Pad(self, onnx_padding, value)
 
 
-@torch_op("aten::contiguous", trace_only=True)
-def aten_contiguous(self: TTensor, memory_format: str = "contiguous_format") -> TTensor:
+@torch_op("aten::contiguous")
+def aten_contiguous(
+    self: TTensor, memory_format: str = "contiguous_format"  # pylint: disable=unused-argument
+) -> TTensor:
     """contiguous(Tensor(a) self, *, MemoryFormat memory_format=contiguous_format) -> Tensor(a)"""
 
-    if memory_format in ["contiguous_format", "preserve_format"]:
-        return op.Identity(self)
-    else:
-        # TODO: Find out a way to annotate constraints for argument, as part of the function meta data structure.
-        raise NotImplementedError(
-            "memory_format value supports 'contiguous_format' or 'preserve_format' only."
-        )
+    # ONNX does not have the notion of memory_format. It is always treated as a no-op.
+    return op.Identity(self)
```

Comment:
I'm wondering if we do so, users calling this function will have a wrong misunderstanding that all of formats were processed successfully, which is not right.
If we can handle this op in an earlier phase by exporter, this should be fine, and we'd better leave a comment here.

Reply:
I understand the format is an inherent representation and does not affect computation in terms of the result? I gathered that from https://pytorch.org/tutorials/intermediate/memory_format_tutorial.html

Reply:
@BowenBao do you have more info on this op? Do you think we should filter it out in a fx pass?